pyrdp/mitm/FileMapping.py

```diff
@@ -82,6 +86,11 @@ def finalize(self):
                 "path": str(self.filesystemPath.relative_to(self.filesystemDir)), "shasum": fileHash
             })
 
+    def onDisconnection(self, reason):
+        if not self.file.closed:
+            self.file.close()
+            Path(self.file.name).unlink(missing_ok=True)
```

Comment:
why are we removing files on disconnection? Is it because they would be incomplete?

Reply:
Yes, if they were complete the `finalize` function would have been called and `self.file.closed` would be `True`.

Reply:
Wouldn't we want to keep partial transfers? Maybe label them as such. Let's say we are monitoring a session where someone uploads a large zip or a VMDK. We could still extract intel from the file even if it's incomplete. Just a thought.